File: backend/src/main/java/com/careermail/controller/AnalyticsController.java
package com.careermail.controller;

import com.careermail.dto.AnalyticsResponse;
import com.careermail.service.AnalyticsService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping({"/api/analytics", "/api/dashboard"})
public class AnalyticsController {

    private final AnalyticsService analyticsService;

    public AnalyticsController(AnalyticsService analyticsService) {
        this.analyticsService = analyticsService;
    }

    @GetMapping({"", "/dashboard", "/summary"})
    public ResponseEntity<AnalyticsResponse> getDashboardAnalytics() {
        return ResponseEntity.ok(analyticsService.getDashboardAnalytics());
    }
}

File: backend/Controllers/AnalyticsController.cs
using CareerMail.Api.Models.DTOs;
using CareerMail.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace CareerMail.Api.Controllers;

[Authorize]
[Route("api/analytics")]
[Route("api/dashboard")]
public class AnalyticsController : BaseApiController
{
    private readonly IAnalyticsService _analyticsService;

    public AnalyticsController(IAnalyticsService analyticsService)
    {
        _analyticsService = analyticsService;
    }

    [HttpGet]
    [HttpGet("dashboard")]
    [HttpGet("summary")]
    public async Task<ActionResult<AnalyticsResponse>> GetDashboardAnalytics()
    {
        var response = await _analyticsService.GetDashboardAnalyticsAsync(CurrentUserId);
        return Ok(response);
    }
}
